openedx/features/course_experience/views/course_home_messages.py

```diff
@@ -55,69 +63,140 @@ def render_to_fragment(self, request, course_id, user_access, **kwargs):
+        )).format(
+            goal_options_container=HTML('<div class="row goal-options-container">')
+        )
+
+        # Add the dismissible option for users that are unsure of their goal
+        goal_choices_html += Text(
+            '{initial_tag}{choice}{closing_tag}'
+        ).format(
+            initial_tag=HTML(
+                '<div tabindex="0" aria-label="{aria_label_choice}" class="goal-option dismissible" '
+                'data-choice="{goal_key}">'
+            ).format(
+                goal_key=CourseGoalOption.UNSURE.value,
+                aria_label_choice=Text(_("Set goal to: {choice}")).format(
+                    choice=get_goal_text(CourseGoalOption.UNSURE.value)
                 ),
-                title=Text('Welcome to {course_display_name}').format(
-                    course_display_name=course.display_name
-                )
+            ),
+            choice=Text(_('{choice}')).format(
+                choice=get_goal_text(CourseGoalOption.UNSURE.value),
+            ),
+            closing_tag=HTML('</div>'),
+        )
+
+        # Add the option to set a goal to earn a certificate,
+        # complete the course or explore the course
+        goal_options = [CourseGoalOption.CERTIFY.value, CourseGoalOption.COMPLETE.value, CourseGoalOption.EXPLORE.value]
+        for goal_key in goal_options:
+            goal_text = get_goal_text(goal_key)
+            goal_choices_html += HTML(
+                '{initial_tag}{goal_text}{closing_tag}'
+            ).format(
+                initial_tag=HTML(
+                    '<div tabindex="0" aria-label="{aria_label_choice}" class="goal-option {col_sel} btn" '
+                    'data-choice="{goal_key}">'
+                ).format(
+                    goal_key=goal_key,
+                    aria_label_choice=Text(_("Set goal to: {goal_text}")).format(
+                        goal_text=Text(_(goal_text))
+                    ),
+                    col_sel='col-' + str(int(math.floor(12 / len(goal_options))))
+                ),
+                goal_text=goal_text,
+                closing_tag=HTML('</div>')
             )
-        if user_access['is_enrolled'] and not course_start_data['already_started']:
-            CourseHomeMessages.register_info_message(
-                request,
-                Text(_(
-                    "Don't forget to add a calendar reminder!"
-                )),
-                title=Text("Course starts in {days_until_start_string} on {course_start_date}.").format(
-                    days_until_start_string=course_start_data['days_until_start_string'],
-                    course_start_date=course_start_data['course_start_date']
-                )
+
+        CourseHomeMessages.register_info_message(
+            request,
+            HTML('{goal_choices_html}{closing_tag}').format(
+                goal_choices_html=goal_choices_html,
+                closing_tag=HTML('</div>')
```

Comment:
Is this an extra closing tag?

Reply:
its not it stems from the unclosed opening tag earlier that starts the course goal messaging well